pilot/pkg/xds/endpoint_builder.go

```diff
@@ -315,9 +323,16 @@ func (b *EndpointBuilder) buildLocalityLbEndpointsFromShards(
 				}
 				localityEpMap[ep.Locality.Label] = locLbEps
 			}
-			if ep.EnvoyEndpoint == nil {
-				ep.EnvoyEndpoint = buildEnvoyLbEndpoint(ep)
+			// TODO(incfly): check with reviewer to see if we should only touches the updated metadata field
+			// or rebuild from scratch (better readability).
+			// if ep.EnvoyEndpoint == nil {
+			tlsMode := ep.TLSMode
+			if b.mtlsChecker != nil && b.mtlsChecker.mtlsDisabledByPeerAuthentication(ep) {
+				log.Infof("incfly pocliy disabled proxy %v, true or false %v",
+					ep.Address, b.mtlsChecker.mtlsDisabledByPeerAuthentication(ep))
+				tlsMode = ""
```

Comment:
just curious, what's the difference between `""` and `model.DisabledTLSModeLabel` for the `tlsMode`?

Reply:
it seems like only used in workload entry and service entry. not much difference in terms of the end effect AFAIK.

Reply:
I think we need to unify "" and DisabledTLSModeLabel, maybe later.